lib/mockito.dart

```diff
@@ -48,6 +53,132 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  return newInvocation;
+}
+
+/// An Invocation implementation that allows all attributes to be passed into
+/// the constructor.
+class FakeInvocation extends Invocation {
```

Comment:
I'm not really in love with this name, and the comment doesn't seem to really apply anymore. To a user they can't pass in all the attributes. Would TypedInvocation be better, or InvocationForTypedArguments? Fake doesn't really tell me much here.